app/flame_chart/profile_model.ts

```diff
@@ -85,8 +90,21 @@ export function buildThreadTimelines(events: TraceEvent[], { visibilityThreshold
       event.dur &&
       event.dur > visibilityThreshold
   );
-  events.sort(eventComparator);
+  
+  for (const [tid, threadName] of threadNameByTid.entries()) {
+    if (tidByThreadName.has(threadName)) {
+      tidByThreadName.set(threadName, Math.min(tidByThreadName.get(threadName), tid));
+    } else {
+      tidByThreadName.set(threadName, tid)
+    }
+  }
+
+  for (const event of events) {
```

Comment:
Maybe a comment above this that says something like:

// Merge all of the threads with a single thread name under the same tid.